medialist.go

```diff
@@ -40,9 +40,17 @@ func (ml *MediaList) AddMedia(m *Media) error {
 	if ml.list == nil {
 		return errors.New("Media list must be initialized first")
 	}
+	err := ml.Lock()
```

Comment:
The docs mention that adding a `Media` to a `MediaList` via `AddMedia` requires locking the `MediaList`... I implemented the lock _within_ the method, which is a departure from how `libvlc` does it (which leaves the locking to the user). Maybe what I did was a mistake! If people read the official docs, they might deadlock? Maybe we should leave Locking to the user like the official docs did.